openvpn/middlewares/client/auth/middleware.go

```diff
@@ -7,25 +7,28 @@ import (
 	"regexp"
 )
 
+// CredentialsProvider returns client's current auth primitives (i.e. customer identity signature / node's sessionId)
+type CredentialsProvider func() (username string, password string, err error)
+
 type middleware struct {
-	authenticator Authenticator
-	connection    net.Conn
-	lastUsername  string
-	lastPassword  string
-	state         openvpn.State
+	credentials  CredentialsProvider
```

Comment:
`credentials` name is a bit confusing - you don't have credentials here, but you have a function which can (in some way) get you credentials. But not always, because it can return you an error as well!

Reply:
renamed to fetchCredentials